packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -58,15 +58,18 @@ class ConfigChainBuilder {
     let foundIgnore = false;
 
     while (loc !== (loc = path.dirname(loc))) {
+      // Find configs in the following file type precedence order:
+      // .babelrc -> .babelrc.js -> package.json
       if (!foundConfig) {
         const configLoc = path.join(loc, BABELRC_FILENAME);
-        if (exists(configLoc)) {
-          this.addConfig(configLoc);
-          foundConfig = true;
-        }
-
+        const configJSLoc = path.join(loc, BABELRC_JS_FILENAME);
         const pkgLoc = path.join(loc, PACKAGE_FILENAME);
-        if (!foundConfig && exists(pkgLoc)) {
+
+        if (exists(configLoc)) {
```

Comment:
@xtuc I agree! I think we should also throw for the package.json If we're going to throw for the additional configuration files. Will have to do a little refactoring to avoid having to double parse the package.json file so we can check its contents.